source/extensions/filters/http/aws_lambda/config.cc

```diff
@@ -1,21 +1,139 @@
 #include "extensions/filters/http/aws_lambda/config.h"
 
+#include "envoy/thread_local/thread_local.h"
+
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace AwsLambda {
 
+namespace CommonAws = Envoy::Extensions::HttpFilters::Common::Aws;
+
+namespace {
+
+// Current AWS implementation will only refresh creds if *more* than an hour has
+// passed. In the AWS impl these creds are fetched on the main thread. As we use
+// the creds in the filter, we cache the creds in a TLS slot, so they are
+// available to the filters in the worker threads. As we have our own cache, we
+// have to implement our own refresh loop. To avoid being to tied down the
+// timing of the AWS implementation, we'll have a refresh interval of a few
+// minutes. refresh itself will only happen when deemed needed by the AWS code.
+//
+// According to the AWS docs
+// (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html)
+//
+// >>> it should get a refreshed set of credentials every hour, or at least 15
+// >>> minutes before the current set expires.
+//
+// Refreshing every 14 minutes should guarantee us fresh credentials.
+constexpr std::chrono::milliseconds REFRESH_AWS_CREDS =
```

Comment:
make this value configurable?

Reply:
not sure it would be useful to make this configurable, as the aws code is hard coded.
we can discuss.